### DISCUSSION TITLE
Can't use ash in composite action

### DISCUSSION THREAD

**Discussion by fortierq** [community/community#26240] [Actions]
<p>I want to use a composite action working on alpine (I plan to use container: image: alpine when calling this action).<br>
However, ash is not allowed as a shell (<a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>). Why ? Is there any workaround ? I guess I can write a Docker-based action instead…</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Two options:</p>
<ol>
<li>Use <code>sh</code>. On Alpine <code>/bin/sh</code> is a symlink to Busybox, and you’ll effectively get <code>ash</code>. This is what I’d do. Works the same, and it’ll work on other Unix-ish platforms, too.</li>
<li>If you want to use <code>/bin/ash</code> explicitly for some reason, scroll down a bit further to the <a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#custom-shell">Custom shell</a> section.</li>
</ol>
